jwst/datamodels/model_base.py

```diff
@@ -170,7 +171,9 @@ def __init__(self, init=None, schema=None, memmap=False,
             asdffile = fits_support.from_fits(init, self._schema, self._ctx,
                                               **kwargs)
 
-        elif isinstance(init, (str, bytes)):
+        elif isinstance(init, (str, bytes, PurePath)):
+            if isinstance(init, PurePath):
+                init = str(init)
```

Comment:
So this is just to allow `Path` input, but internally, everything is still a path string?

Is it worth trying to use `Path` objects throughout the package and converting string paths into `Path` objects?  Or is that too much to bite off for now?

Reply:
It probably is worth it, but that should have a design discussion and an issue to update the internals.

Practically, I did start down the rabbit-hole but, on the third module, made the executive decision to abort. The goal was to allow the user-facing API to handle Path. Updating the internals is a different task.

Reply:
Agree.  Going through this exercise though would catch instances where we are assuming filesystems are unix (vs Windows), and would allow us to support Windows with simpler code.  See #4996.